### PR TITLE
improved the way of displaying the doc links

### LINKED ISSUE #1526: More General Method of Displaying Docs Links
### Please describe the problem (or idea)
The following code is used to display the docs links for each module in the UI.
https://github.com/publiclab/image-sequencer/blob/d1b9b8beccdbe5ba98cf4e352af103b4913a5fe6/examples/lib/defaultHtmlStepUi.js#L48-L50

The code just links to a general link that may not work for every module. Instead, the link should be imported from every module's `info.json` file instead.
https://github.com/publiclab/image-sequencer/blob/d1b9b8beccdbe5ba98cf4e352af103b4913a5fe6/src/modules/BlobAnalysis/info.json#L6

### Please show us where to look

https://beta.sequencer.publiclab.org

### What's your PublicLab.org username?

> This can help us diagnose the issue:

### Browser, version, and operating system

> Many bugs are related to these -- please help us track it down and reproduce what you're seeing!

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site.

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by vivek-30**
Fixes #1526 .
As mentioned in the issue i have used docs-link straight from their respective modules instead of hard coding them.
so this provides a more general approach to represent these docs-link.

### CODE REVIEW

File: examples/lib/defaultHtmlStepUi.js
Comment (HarshKhandeparkar) [inline examples/lib/defaultHtmlStepUi.js:29]: You can just do `step['docs-link']` here, no `require` required. (And also not allowed in browsers)

File: examples/lib/defaultHtmlStepUi.js
Comment (HarshKhandeparkar) [inline examples/lib/defaultHtmlStepUi.js:54]: The href will need quotes here
`href="' + packageLink + '>'`

Also, spaces between the `+` and other things will make it look cleaner.

**Review by HarshKhandeparkar — CHANGES_REQUESTED** [review]
Nice work. Just a few small changes required :)

File: examples/lib/defaultHtmlStepUi.js
Comment (HarshKhandeparkar) [inline examples/lib/defaultHtmlStepUi.js:28]: Can we use `const` here or just write `steps['docs-link']` directly below as this variable is used only once?

Comment (vivek-30) [reply examples/lib/defaultHtmlStepUi.js:28]: sure @HarshKhandeparkar ,it will improve code  maintainability

Comment (HarshKhandeparkar) [reply examples/lib/defaultHtmlStepUi.js]: Thank you!

**Comment by HarshKhandeparkar** [comment]
Can you show a screenshot/GIF of this working? (It showed an error on gitpod)

**Comment by HarshKhandeparkar** [comment]
Sorry, it's not your mistake. I thought `steps['docs-link']` will work :sweat_smile:

**Comment by HarshKhandeparkar** [comment]
Don't worry, I am pushing a fix :)

**Comment by HarshKhandeparkar** [comment]
Some of the backend is quite messy :sweat_smile:

**Comment by HarshKhandeparkar** [comment]
I have no idea how to make this work :joy:

**Comment by vivek-30** [comment]
so i think the main problem is that -
changes that took place looks like -
```diff
-  '<a href="https://github.com/publiclab/image-sequencer/blob/main/docs/MODULES.md#' + step.name + '-module">' 
+  '<a href="' + step['docs-link'] + '">'

```
so there is no problem in old code bcoz we are just hard coding the link and only provide `step.name` which is fine as name property is available in every file but when i went through each and every module then this is what i have found

there are 3 kinds of file :
- some have `docs-link` property completely empty.  (modules - constrainedCrop,gridOverlay,noiseReduction,textOverlay. )

- some don't have `module` appended at their end (i.e completed partially)  (modules - blobAnalysis, colorTemperature, mask, rotate, threshold, tint. )

- and all other have complete link 

so if you allow then i will write `docs-link` in files where ever it requires.

but one more thing each module directory name follow camel case but their name in `info.json` file are seperated with hyphens so how to tackle this any suggestions .
BTW this issue is a bit tricky then it seems 😅

**Comment by HarshKhandeparkar** [comment]
It seems that not all properties from `info.json` are passed on to the `step` variable and currently I have no idea how to fix it :sweat:

**Comment by HarshKhandeparkar** [comment]
@vivek-30 Here's a fix: convert `step['docs-link']` to `step.moduleInfo['docs-link']` :)

**Comment by vivek-30** [comment]
ok. Thanks a lot for your kind help 😊  . let me try this, hope it works .

File: examples/lib/defaultHtmlStepUi.js
Comment (HarshKhandeparkar) [inline examples/lib/defaultHtmlStepUi.js:51]:
```suggestion
                    '<a href="' + step.moduleInfo['docs-link'] + '">' + (step.description || '') + '</a>' +
```

Tiny mistake

**Review by HarshKhandeparkar — CHANGES_REQUESTED** [review]
Small typo :)

**Comment by vivek-30** [comment]
Is it working now ? Also  @HarshKhandeparkar  do i close this PR now don't know what to do

**Comment by HarshKhandeparkar** [comment]
1. Yes it is working :tada:.
2. You just have to wait until a maintainer merges it, you don't have to do anything :slightly_smiling_face:.

**Review by HarshKhandeparkar — APPROVED** [review]
:tada:

File: examples/lib/defaultHtmlStepUi.js
Comment (jywarren) [inline examples/lib/defaultHtmlStepUi.js:29]: Hi all!!!  Should we default to https://github.com/publiclab/image-sequencer/blob/main/docs/MODULES.md do you think?

Comment (HarshKhandeparkar) [reply examples/lib/defaultHtmlStepUi.js:29]: We can do that! Does the first step have any docs anyway?

Comment (jywarren) [reply examples/lib/defaultHtmlStepUi.js:29]: Maybe not but it could have a brief explanation maybe, we can add it later. Thanks!!!

Comment (HarshKhandeparkar) [reply examples/lib/defaultHtmlStepUi.js:29]: @vivek-30 want to try changing the default link to https://github.com/publiclab/image-sequencer/blob/main/docs/MODULES.md?

Comment (vivek-30) [reply examples/lib/defaultHtmlStepUi.js:29]: @HarshKhandeparkar and @jywarren  i have fixed another issue #1724 , and also tested it. so  can i raise it with some different branch ?

Comment (HarshKhandeparkar) [reply examples/lib/defaultHtmlStepUi.js:29]: Yes

Comment (HarshKhandeparkar) [reply examples/lib/defaultHtmlStepUi.js:29]: Use a different branch and a separate PR.

**Comment by vivek-30** [comment]
sorry sir i am just renaming the branch and did not knew that it will close this PR 😥  . That was not my intention , so should i reopen this PR or leave it closed? please suggest me.

**Comment by HarshKhandeparkar** [comment]
You can open a new PR :)

**Comment by vivek-30** [comment]
Ok Thanks 😊

**Pull request closed without merging**
